src/commands/signzone.rs

```diff
@@ -1360,14 +1362,29 @@ impl SignZone {
         // bytes than are needed. Instead control the allocation size based on
         // our knowledge of the file size.
         let mut zone_file = File::open(zonefile_path)
-            .map_err(Error::from)
+            .map_err(|e| format!("error opening file: {e}").into())
             .context(&format!(
                 "loading zone file from path '{}'",
                 zonefile_path.display(),
             ))?;
-        let zone_file_len = zone_file.metadata()?.len();
+        let zone_file_len = zone_file
+            .metadata()
+            .map_err::<Error, _>(|e| {
+                format!(
+                    "error getting metadata from zonefile {}: {e}",
+                    zonefile_path.display()
+                )
+                .into()
+            })?
```

Comment:
I think this should work too (removing both the type annotation and the `into`)
```suggestion
            .map_err(|e| {
                format!(
                    "error getting metadata from zonefile {}: {e}",
                    zonefile_path.display()
                )
            })?
```